cmd/otel-agent/config/agent_config.go

```diff
@@ -75,6 +105,22 @@ func NewConfigComponent(ctx context.Context, ddCfg string, uris []string) (confi
 		}
 	}
 
+	// Set the right log level. The most verbose setting takes precedence.
+	var logLevel string
+	agentLogLevel := pkgconfig.GetString("log_level")
+	telemetryLogLevel := sc.Telemetry.Logs.Level
+	agentLogMapping, ok := logLevelMap[agentLogLevel]
+	if ok {
```

Comment:
I would rather return an error here since I don't think `ok` should ever be false. If it is false, the mapping is incomplete and we should update.

Reply:
thanks, I think this ties into my comment above and is a better approach.